catalog/camel-report-maven-plugin/src/main/java/org/apache/camel/maven/RouteCoverageMojo.java

```diff
@@ -319,8 +332,16 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             }
         }
 
+        // compute and log overall coverage across routes
+        AtomicBoolean overallCoverageAboveThreshold = new AtomicBoolean();
+        String out = templateOverallCoverageData(coveredNodes.get(), totalNumberOfNodes, overallCoverageAboveThreshold);
+        getLog().info("Overall coverage summary:\n\n" + out);
+        getLog().info("");
+
         if (failOnError && notCovered.get() > 0) {
             throw new MojoExecutionException("There are " + notCovered.get() + " route(s) not fully covered!");
+        } else if (failOnError && !overallCoverageAboveThreshold.get()) {
+            throw new MojoExecutionException("The overall coverage is below " + overallCoverageThreshold + "%!");
```

Comment:
what about providing the current value in the message? It will allow to see rigth away if it is far from the objective or not.

Reply:
Good point. I didn't include the coverage value here, since one line above it, I already print the coverage anyway (see my other comment). But if e.g. your IDE only shows you the error message, it could be a good addition.